### ISSUE TITLE
Prototype of processing pipeline

### ISSUE THREAD

**Issue by jcohenadad** [sct-pipeline/critical-ms-lesion#3]
Let's get started with the dummy dataset #1 

subsequent issues:

- [x] - segment lesion
- [x] - normalize CSA with PAM50 (see work of Sandrine and Jan)
- [ ] - replace vertebral labeling with TotalSpineSeg
- [ ] - compute lesion occupancy within lateral CST
- [ ] - define strategy to compute local atrophy at lesion levels (consider other metrics than CSA)

**Comment by plbenveniste** [comment]
I investigated the detection of local atrophy in the spinal cord on the data that we have received. I was planning on using the following method: 
- if locally the CSA or the R-L diameter is less than 80% of the above and lower vertebral level, then there is local atrophy.

However, as we can see on the figure below, because there is quite a lot of variability of both the CSA and the R-L diameter, this method wouldn't work. 

![plot1](https://github.com/user-attachments/assets/8b8e0365-5da7-4f42-9427-abe598387a27)
![plot2](https://github.com/user-attachments/assets/92ecd103-b988-4c5e-9a70-5da136813e83)

Moving on, atrophy should be detected by measuring deviance between the subject CSA and/or R-L diameter and average CSA or R-L diameter of a population. 
Some questions emerge from this: 
- Do we have a template/atlas of the evolution of the CSA and/or R-L diameter?
- If not, we should build one. Should we use the data given in this project in order to work on the same contrast ? 
- Should we differentiate men and women ? 

Maybe these questions could be raised during next meeting.

**Comment by jcohenadad** [comment]
> Moving on, atrophy should be detected by measuring deviance between the subject CSA and/or R-L diameter and average CSA or R-L diameter of a population.
Some questions emerge from this:
Do we have a template/atlas of the evolution of the CSA and/or R-L diameter?

Hence my comment above "see work of Sandrine and Jan" 😉  have a chat with @valosekj @sandrinebedard, they can show you how

**EDIT 20241125_211019**: Ah! It seems like you are already aware:
https://github.com/sct-pipeline/critical-ms-lesion/blob/028b073cba7e6e3fc7d37ecccff135cb5d7874a2/batch_processing.sh#L168

**Comment by valosekj** [comment]
Nice figures, @plbenveniste. Just fyi, if you need to show both PAM50 slices and vert levels on a single x-axis, you could use [this script](https://github.com/valosekj/dcm-brno/blob/main/03_plotting_scripts/02b_generate_figures_PAM50_two_sessions.py), or [its parts](https://github.com/valosekj/dcm-brno/blob/main/03_plotting_scripts/02b_generate_figures_PAM50_two_sessions.py#L269-L285). It generates the following figure:

<img width="570" alt="image" src="https://github.com/user-attachments/assets/bbbe98b0-2d85-496d-9ea2-006ed1e7763f">

The figure shows single-subject morphometrics for two sessions. Together with normative values computed from spine-generic (see [this repo](https://github.com/spinalcordtoolbox/PAM50-normalized-metrics/tree/main) for one CSV file for each subject from the spine-generic dataset). The normative values can be [filtered by sex](https://github.com/valosekj/dcm-brno/blob/99e287d3db96ae6d77c80546154c3737cb686dfe/03_plotting_scripts/02b_generate_figures_PAM50_two_sessions.py#L78).

---

what is the ref for the method you mentioned?

> if locally the CSA or the R-L diameter is less than 80% of the above and lower vertebral level, then there is local atrophy.

**Comment by plbenveniste** [comment]
Thanks for your comment @valosekj. If I understand correctly, you don't have an average CSV file of the entire cohort. You only have one CSV file per subject. 
So if I were to use your data, I would have to get all CSV files, average the CSA or R-L diameter per sex so that I could use it as a standard reference. Is that correct ? 

> what is the ref for the method you mentioned?

No method in particular, just my idea of how to solve this. It proved to be a failure.

**Comment by sandrinebedard** [comment]
> So if I were to use your data, I would have to get all CSV files, average the CSA or R-L diameter per sex so that I could use it as a standard reference. Is that correct ?
> 
> >

yes, we provide it per subject, to allow custom filtering (like per sex),

this is what we use in `sct_compute_compression -normalize-hc 1` (https://spinalcordtoolbox.com/stable/user_section/command-line/sct_compute_compression.html)

**Comment by valosekj** [comment]
> So if I were to use your data, I would have to get all CSV files, 

"all CSV files" from the [PAM50-normalized-metrics](https://github.com/spinalcordtoolbox/PAM50-normalized-metrics) are automatically downloaded during SCT installation (to `$SCT_DIR/data/PAM50_normalized_metrics` folder) because we use them for `sct_compute_compression` as mentioned by Sandrine.
You can load all the CSV files into a single dataframe using [this function](https://github.com/valosekj/dcm-brno/blob/99e287d3db96ae6d77c80546154c3737cb686dfe/03_plotting_scripts/02b_generate_figures_PAM50_two_sessions.py#L86).

> average the CSA or R-L diameter per sex so that I could use it as a standard reference. Is that correct ?

Once you load all the CSV files into a single dataframe, you can plot them as mean and std directly using `sns.lineplot errorbar='sd'`, see [this line](https://github.com/valosekj/dcm-brno/blob/99e287d3db96ae6d77c80546154c3737cb686dfe/03_plotting_scripts/02b_generate_figures_PAM50_two_sessions.py#L230C96-L230C109). i.e., no need to average them manually
